vcd/resource_vcd_vdc.go

```diff
@@ -0,0 +1,505 @@
+				},
+				"reserved": {
+					Type:        schema.TypeInt,
+					Optional:    true,
+					ForceNew:    true,
+					Description: "Capacity reserved",
+				},
+				"used": {
+					Type:        schema.TypeInt,
+					Optional:    true,
+					ForceNew:    true,
+					Description: "Capacity used. If the VDC AllocationModel is ReservationPool, this number represents the percentage of the reservation that is in use. For all other allocation models, it represents the percentage of the allocation that is in use.",
+				},
+				"overhead": {
+					Type:        schema.TypeInt,
+					Optional:    true,
+					ForceNew:    true,
+					Description: "Number of Units allocated to system resources such as vShield Manager virtual machines and shadow virtual machines provisioned from this Provider VDC.",
+				},
+			},
+		},
+	}
+
+	return &schema.Resource{
+		Create: resourceVcdVdcCreate,
+		Delete: resourceVcdVdcDelete,
+		Read:   resourceVcdVdcRead,
+
+		Schema: map[string]*schema.Schema{
+			"org": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "Organization to create the VDC in",
+			},
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"description": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"allocation_model": &schema.Schema{
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringInSlice([]string{"AllocationVApp", "AllocationPool", "ReservationPool"}, false),
+				Description:  "The allocation model used by this VDC; must be one of {AllocationVApp, AllocationPool, ReservationPool}",
+			},
+			"compute_capacity": &schema.Schema{
+				Required: true,
+				ForceNew: true,
+				Type:     schema.TypeSet,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"cpu":    &capacityWithUsage,
+						"memory": &capacityWithUsage,
+					},
+				},
+				Description: "The compute capacity allocated to this VDC.",
+			},
+			"nic_quota": &schema.Schema{
+				Type:        schema.TypeInt,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "Maximum number of virtual NICs allowed in this VDC. Defaults to 0, which specifies an unlimited number.",
+			},
+			"network_quota": &schema.Schema{
+				Type:        schema.TypeInt,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "Maximum number of network objects that can be deployed in this VDC. Defaults to 0, which means no networks can be deployed.",
+			},
+			"vm_quota": &schema.Schema{
+				Type:        schema.TypeInt,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "The maximum number of VMs that can be created in this VDC. Includes deployed and undeployed VMs in vApps and vApp templates. Defaults to 0, which specifies an unlimited number.",
+			},
+			"is_enabled": &schema.Schema{
```

Comment:
If enabled is the normal condition of a newly created VDC, we need a property to disable it.
What happens is we omit this field? It is not clear. Perhaps adding a field `disable_on_creation` would be better.
It's the same problem we have with `over_commit_allowed` below

Reply:
Having negative logic in the parameter names is even more confusing. The description says that "A VDC is always enabled on creation". Is this actual behaviour? If it is, then can we just use the "Default" word before to make it really clear and consistent with other descriptions? Something like `Default is true, which ebables VDC on creation`.

> What happens is we omit this field?

Should be enabled by default.

>  we need a property to disable it

`is_enabled = "false"` should work.

But let's see what @vbauzysvmware replies.

Reply:
My point here is that if we omit `is_enabled`, its value will be false, but the VDC will be enabled by default (i.e. the behavior when we don't set an optional property).

Reply:
we agreed to change to `enabled`, default value configured, comments improved.